### PR TITLE
Make Doxygen comment style the same

### PR DESCRIPTION
**Pull request by k-doering-NOAA**
# What is the feature?

* Use `/**` (Javadoc style) as the consistent style for doxygen

# How have you implemented the solution?

Modified other versions (including `/*!` and `/**********`)
There was [one file](https://github.com/NOAA-FIMS/FIMS/blob/fe84d2f717cb0752beed8cefce23df18b9faba1a/inst/include/utilities/fims_json.hpp#L1) that I wasn't sure if I should change to using `//` for c++ comments and `/**` for doxygen comments.

# Does the PR impact any other area of the project?

No

# How to test this change

Check github actions pass and documentation can be built.

# Developer pre-PR checklist

- [x] I relied on GitHub actions to :test_tube: things for me while I sat on the :couch_and_lamp:.

### CODE REVIEW

File: inst/include/interface/rcpp/rcpp_objects/rcpp_growth.hpp [hunk 1]
```diff
@@ -78,7 +78,7 @@ vectors have been set */
    * @param ages Type vector of ages
    * @return std::map<T, T>
    *
-   * */
+   */
```
Comment (Bai-Li-NOAA) [inline inst/include/interface/rcpp/rcpp_objects/rcpp_growth.hpp:81]: There are a few `**/` in the codebase (e.g., under FIMS/inst/include/interface/rcpp/rcpp_objects/ folder and fleet.hpp). Not sure if you want to change all of them to `*/`.

**Review by Bai-Li-NOAA — COMMENTED** [review]
Thanks for making all those changes. There are a few `\*!` in the F`IMS/vignettes/fims-path-maturity.Rmd` that may need to be updated as well.

**Comment by k-doering-NOAA** [comment]
@Bai-Li-NOAA thanks so much! I wasn't sure about whether to change the FIMS_json file, so thank you for taking care of that.

I fixed the other 2 issues that you identified (some `**/` that should be `*/` and `/*!` to `/**` in the vignette.

Let me know if you think any other changes need to be made!

**Review by Bai-Li-NOAA — APPROVED** [review]
Looks great!

**Pull request merged**